src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java

```diff
@@ -221,24 +249,29 @@ public void testEmbeddedFieldList() throws Exception {
     InitCodeNode innerList = InitCodeNode.create("0");
     InitCodeNode innerStructure =
         InitCodeNode.createWithChildren("mylist", InitCodeLineType.ListInitLine, innerList);
-    InitCodeNode expectedStructure =
+    InitCodeNode outerStructure =
         InitCodeNode.createWithChildren(
             "myfield", InitCodeLineType.StructureInitLine, innerStructure);
 
-    InitCodeNode actualStructure = FieldStructureParser.parse(fieldSpec);
-    Truth.assertThat(checkEquals(actualStructure, expectedStructure)).isTrue();
+    InitCodeNode expectedStructure = InitCodeNode.newRoot();
+    expectedStructure.mergeChild(outerStructure);
+
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(actualStructure, fieldSpec);
+
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void testFormattedFieldBadField() throws Exception {
     String fieldSpec = "name%entity";
-    FieldStructureParser.parse(fieldSpec);
+    FieldStructureParser.parse(InitCodeNode.newRoot(), fieldSpec);
```

Comment:
Could we add a comment to this and the following saying the error is because no `InitValueConfig` containing `name` was provided?

Reply:
Done